### PR TITLE
Add generics to OutputBlockData

### PR DESCRIPTION
**Pull request by hata6502**
https://github.com/codex-team/editor.js/issues/1551

### CODE REVIEW

File: types/data-formats/output-data.d.ts [hunk 1]
```diff
@@ -3,15 +3,15 @@ import {BlockToolData} from '../tools';
 /**
  * Output of one Tool
  */
-export interface OutputBlockData {
+export interface OutputBlockData<S extends string = string, T extends object = any> {
   /**
    * Too type
```
Comment (sosie-js) [inline types/data-formats/output-data.d.ts:8]: Tool type

Comment (hata6502) [reply types/data-formats/output-data.d.ts]: Thanks!

**Comment by hata6502** [comment]
I create a repository to introduce the better way to use Editor.js with TypeScript. 
Please see: https://github.com/hata6502/editorjs-examples/blob/main/examples/EditorJSData.ts

File: types/data-formats/output-data.d.ts [hunk 2]
```diff
@@ -3,15 +3,15 @@ import {BlockToolData} from '../tools';
...
  * Output of one Tool
  */
-export interface OutputBlockData {
+export interface OutputBlockData<S extends string = string, T extends object = any> {
   /**
-   * Too type
+   * Tool type
    */
-  type: string;
+  type: S;
```
Comment (neSpecc) [inline types/data-formats/output-data.d.ts:10]: The type is always a string. Are you sure we need a generic here?

**Review by neSpecc — CHANGES_REQUESTED** [review]

File: types/data-formats/output-data.d.ts [hunk 3]
```diff
@@ -3,15 +3,15 @@ import {BlockToolData} from '../tools';
 /**
  * Output of one Tool
  */
-export interface OutputBlockData {
+export interface OutputBlockData<S extends string = string, T extends object = any> {
```
Comment (neSpecc) [inline types/data-formats/output-data.d.ts:6]:
```suggestion
export interface OutputBlockData<S extends string = string, Data extends object = any> {
```

Comment (neSpecc) [reply types/data-formats/output-data.d.ts:6]: and describe it via `@template` please

**Review by neSpecc — CHANGES_REQUESTED** [review]

File: types/data-formats/output-data.d.ts (hunk 2 above)
Comment (hata6502) [reply types/data-formats/output-data.d.ts:10]: By using Generics, Editor.js users can describe string literal types. 

https://www.typescriptlang.org/docs/handbook/literal-types.html#string-literal-types

By using string literal types in `OutputBlockData['type']`, Editor.js users can describe more type-safe `OutputData` in each application. 

See also: 
https://github.com/hata6502/editorjs-examples/blob/main/examples/ExampleEditorJSData.ts#L17

**Review by neSpecc — APPROVED** [review]

**Review by gohabereg — APPROVED** [review]

**Pull request merged**